1-js/04-object-basics/02-garbage-collection/article.md

```diff
@@ -115,98 +115,98 @@ delete family.mother.husband;
 
-After garbage collection:
+După colectarea reziduurilor:
 
 ![](family-no-father-2.svg)
 
-## Unreachable island
+## Insula inaccesibilă
 
-It is possible that the whole island of interlinked objects becomes unreachable and is removed from the memory.
+Este posibil ca toată insula cu obiectele interconectate să devină inaccesibilă și să fie eliminată din memorie.
 
-The source object is the same as above. Then:
+Obiectul sursă este la fel ca mai sus. Apoi:
 
 ```js
 family = null;
 ```
 
-The in-memory picture becomes:
+Imaginea din memorie devine:
 
 ![](family-no-family.svg)
 
-This example demonstrates how important the concept of reachability is.
+Acest exemplu demonstrează cât de important este conceptul de accesibilitate.
 
-It's obvious that John and Ann are still linked, both have incoming references. But that's not enough.
+Este evident că John și Ann sunt încă legate, amăndouă au referințe de intrare. Dar nu este suficient.
 
-The former `"family"` object has been unlinked from the root, there's no reference to it any more, so the whole island becomes unreachable and will be removed.
+Fostul obiect `"family"` a fost deconectat de la rădăcină, nu mai există nicio referință la el, deci întreaga insulă devine iaccesibilă și va fi eliminată.
 
-## Internal algorithms
+## Algoritmi interni
 
-The basic garbage collection algorithm is called "mark-and-sweep".
+Algoritmul de bază de colectare a reziduurilor este denumit "mark-and-sweep" (în traducere literală "marchează-și-mătură").
 
-The following "garbage collection" steps are regularly performed:
+Următorii pași ai "colectării reziduurilor" sunt efectuați regulat:
 
-- The garbage collector takes roots and "marks" (remembers) them.
-- Then it visits and "marks" all references from them.
-- Then it visits marked objects and marks *their* references. All visited objects are remembered, so as not to visit the same object twice in the future.
-- ...And so on until there are unvisited references (reachable from the roots).
-- All objects except marked ones are removed.
+- "Garbage collector" ia rădăcinile și le "marchează" (le memorează).
+- Apoi vizitează toate referințele dinspre ele.
+- Apoi parcurge obiectele marcate și însemnează referințele *lor*. Toate obiectele vizitate sunt memorate pentru a nu vizita același obiect de două ori în viitor.
+- ...Și așa mai departe până când nu există referințe nevizitate (accesibil din rădăcini).
+- Toate obiectele, cu excepția celor marcate, sunt eliminate.
 
-For instance, let our object structure look like this:
+De exemplu, să presupunem că structura obiectului nostru arată așa:
 
 ![](garbage-collection-1.svg)
 
-We can clearly see an "unreachable island" to the right side. Now let's see how "mark-and-sweep" garbage collector deals with it.
+Putem vedea foarte clar o "insulă inaccesibilă" în partea dreaptă. Acum să vedem cum se ocupă colectorul de reziduuri de "mark-and-sweep".
```

Comment:
Grammar suggestion : "Acum să vedem cum se ocupă colectorul de reziduuri de "mark-and-sweep"." -> " Acum să vedem cum se ocupă colectorul de reziduuri "mark-and-sweep" de aceasta."